### PR TITLE
ICU-20022 Fix quarter issue in RelativeDateTimeFormatter

### PR DESCRIPTION
**Pull request by FrankYFTang**
https://unicode-org.atlassian.net :  ICU-20022

### CODE REVIEW

**Comment by jungshik** [comment]
@sffc can you approve this?

**Comment by FrankYFTang** [comment]
ping

**Comment by sffc** [comment]
The process is that you need to set someone as "assignee" in order to get a review.  You could set me, Markus, Andy, etc.

**Comment by FrankYFTang** [comment]
The UI does not let me to change any reviewer or assignee

**Comment by sffc** [comment]
Ok, I'm not sure what the permissions need to be but I put Markus as the assignee for you.

File: icu4c/source/i18n/reldatefmt.cpp [hunk 1]
```diff
@@ -131,12 +135,33 @@ const UnicodeString& RelativeDateTimeCacheData::getAbsoluteUnitString(
...
+   switch (unit) {
+       case UDAT_RELATIVE_YEARS:   rdtunit = UDAT_REL_UNIT_YEAR; break;
+       case UDAT_RELATIVE_MONTHS:  rdtunit = UDAT_REL_UNIT_MONTH; break;
+       case UDAT_RELATIVE_WEEKS:   rdtunit = UDAT_REL_UNIT_WEEK; break;
+       case UDAT_RELATIVE_DAYS:    rdtunit = UDAT_REL_UNIT_DAY; break;
+       case UDAT_RELATIVE_HOURS:   rdtunit = UDAT_REL_UNIT_HOUR; break;
+       case UDAT_RELATIVE_MINUTES: rdtunit = UDAT_REL_UNIT_MINUTE; break;
+       case UDAT_RELATIVE_SECONDS: rdtunit = UDAT_REL_UNIT_SECOND; break;
+       default: // a unit that the above method does not handle
+            return NULL;
```
Comment (markusicu) [inline icu4c/source/i18n/reldatefmt.cpp]: Optional: Now that we are on C++11, we want to use nullptr in new code.

File: icu4c/source/i18n/unicode/reldatefmt.h [hunk 2]
```diff
@@ -165,6 +165,12 @@ typedef enum UDateAbsoluteUnit {
      */
     UDAT_ABSOLUTE_NOW,
 
+    /**
```
[the remark this answers is not in the document]
Comment (FrankYFTang) [reply icu4c/source/i18n/unicode/reldatefmt.h]: done

File: icu4c/source/i18n/reldatefmt.cpp (hunk 1 above)
Comment (FrankYFTang) [reply icu4c/source/i18n/reldatefmt.cpp]: done

**Comment by FrankYFTang** [comment]
PTAL

**Pull request merged**